app/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -127,8 +142,13 @@ protected DateTime getFirstEventDateTime(ScheduleContext context, String eventId
      * If scheduling hasn't reached the end time, or hasn't accumulated the minimum number of tasks, returns true, or 
      * false otherwise. 
      */
-    protected boolean shouldContinueScheduling(ScheduleContext context, DateTime scheduledTime, List<ScheduledActivity> scheduledActivities) {
-        boolean boundaryNotMet = scheduledTime.isBefore(context.getEndsOn()) || hasNotMetMinimumCount(context, scheduledActivities.size());
+    protected boolean shouldContinueScheduling(ScheduleContext context, DateTime scheduledTime,
+            List<ScheduledActivity> scheduledActivities) {
+        DateTime scheduledTimeInZone = scheduledTime.withZone(context.getZone());
+        DateTime endsOnInZone = context.getEndsOn().withZone(context.getZone());
+        
+        boolean boundaryNotMet = scheduledTimeInZone.isBefore(endsOnInZone) || 
+                hasNotMetMinimumCount(context, scheduledActivities.size());
```

Comment:
Given that our timezones are using offsets and not names like "PST", I believe this change is actually not necessary. Two datetimes will be in the same relative order no matter what offset your convert them to, so forcing them into the user's local timezone doesn't actually change anything.